config.h.guess

```diff
@@ -853,5 +853,33 @@
 #define STRERROR_R_CHAR_P ((AG_GLIBC_PREREQ(0,0) || AG_UCLIBC_PREREQ(0,0,0)) && (HAVE__GNU_SOURCE || !(_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600)))
 #endif
 
+#ifndef HAVE_GLIBC_PTHREAD
```

Comment:
The define is only really about pthread_setname, please rename them all accordingly

Reply:
Different pthread implementations have different function signatures, and some only have `setname` and no `getname`. That's why I've introduced the `HAVE_<PLATFORM>_PTHREAD` macro as welll as `HAVE_PTHREAD_SETNAME` and `HAVE_PTHREAD_GETNAME`.

Reply:
I don't think that really makes sense though: the values you've chosen here are very geared to the setname call.
e.g. glibc had pthreads long before 2.12.